src/topology/category/Top/limits.lean

```diff
@@ -101,3 +102,96 @@ instance forget_preserves_colimits : preserves_colimits (forget : Top.{u} ⥤ Ty
       (colimit_cocone_is_colimit F) (types.colimit_cocone_is_colimit (F ⋙ forget)) } }
 
 end Top
+
+namespace Top
+
+section topological_konig
+
+/-!
+## Topological Kőnig's lemma
+
+A topological version of Kőnig's lemma is that the inverse limit of nonempty compact Hausdorff
+spaces is nonempty.  (Note: this can be generalized further to inverse limits of nonempty compact
+T0 spaces, where all the maps are closed maps; see A.H. Stone, "Inverse limits of compact spaces").
+-/
+
+variables {J : Type u} [directed_order J]
+variables (F : Jᵒᵖ ⥤ Top.{u})
+
+/--
+The partial sections of an inverse system of topological spaces from an index `j` are sections
+when restricted to all objects less than or equal to `j`.
+-/
+def partial_sections (j : Jᵒᵖ) : set (Π j, F.obj j) :=
+{ u | ∀ {j'} (f : j ⟶ j'), F.map f (u j) = u j'}
+
+lemma partial_sections.nonempty [Π (j : Jᵒᵖ), nonempty (F.obj j)] (j : Jᵒᵖ) :
+  (partial_sections F j).nonempty :=
+begin
+  haveI := classical.dec_pred (λ (j' : Jᵒᵖ), j'.unop ≤ j.unop),
+  use λ (j' : Jᵒᵖ),
+    if h : j'.unop ≤ j.unop then
+      F.map (hom_of_le h).op (classical.arbitrary (F.obj j))
+    else
+      classical.arbitrary _,
+  intros j' fle,
+  simp only [le_of_hom fle.unop, dif_pos],
+  erw category_theory.functor.map_id,
+  rw (by dec_trivial : fle = (hom_of_le (le_of_hom fle.unop)).op),
```

Comment:
These cumbersome steps (the `erw`, and the `by dec_trivial`) are only necessary because we're missing some lemmas about `hom_of_le` and `le_of_hom`, and about `iff_true_intro`.

I've taken the liberty of pushing a commit with these additional lemmas, because they belong in two different files.